main/EDInitMod.F90

```diff
@@ -803,15 +807,21 @@ subroutine init_cohorts( site_in, patch_in, bc_in)
              use_pft_local(pft) = ifalse ! Case 4
           endif
        endif
-
     end do
+
+    
     do pft =  1,numpft
+
        if(use_pft_local(pft).eq.itrue)then
-          if(EDPftvarcon_inst%initd(pft)>nearzero) then
 
-             allocate(temp_cohort) ! temporary cohort
+          allocate(temp_cohort) ! temporary cohort
+          temp_cohort%pft         = pft
+          temp_cohort%l2fr = prt_params%allom_l2fr(pft)
+          temp_cohort%canopy_trim = 1.0_r8
+          temp_cohort%crowndamage = 1  ! Assume no damage to begin with
+
+          if(EDPftvarcon_inst%initd(pft)>nearzero) then   ! interpret as initial density and calculate diameter
```

Comment:
@JessicaNeedham Should we gracefully fail (during parameter reading) a cold-start simulation where initd=0 (ie abs(initd)<nearzero)  ?  It seems like if someone sets the diameter to zero, or the density to zero, the run will fail anyway.  Although the user may be wondering what happened.

Reply:
Good idea. I've added a check in EDPftvarcon